forward/fwd_websocket_test.go

```diff
@@ -150,6 +149,52 @@ func (s *FwdSuite) TestWebsocketRequestWithOrigin(c *C) {
 	c.Assert(resp, Equals, "ok")
 }
 
+func (s *FwdSuite) TestWebsocketRequestWithQueryParams(c *C) {
+	f, err := New()
+	c.Assert(err, IsNil)
+
+	upgrader := gorillawebsocket.Upgrader{}
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		conn, err := upgrader.Upgrade(w, r, nil)
+		if err != nil {
+			return
+		}
+		defer conn.Close()
+		c.Assert(r.URL.Query().Get("query"), Equals, "test")
+		for {
```

Comment:
In fact, the ReadMessage block until a message is received or the connection is closed so, it will not consume CPU etc...